### PR TITLE
Rebuild with python 3.10 and cross compile aarch64+ppc

### PR DESCRIPTION
**Pull request by akrherz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
*  Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Trying the suggested cross-compile + python 3.10 from #292

### CODE REVIEW

**Comment by akrherz** [comment]
appears the error is 

> 2021-12-02T19:32:19.2277685Z + test -f /home/conda/feedstock_root/build_artifacts/libopencv_1638471684005/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libopencv_hdf.so
2021-12-02T19:32:19.6979622Z Tests failed for libopencv-4.5.3-py37h533c580_6.tar.bz2

So hdf support did not get built in? ... trying something...

**Comment by akrherz** [comment]
@conda-forge-admin please restart ci

**Comment by hmaarrfk** [comment]
Maybe try adding to `build` section;
```
- hdf5 # [build_platform != target_platform]
```

but tbh, I can't find `WITH_HDF` anywhere in the source. Nor can I find `HDF` in any `CMakeLists.txt` file. So i'm just confused.

**Comment by akrherz** [comment]
Thanks @hmaarrfk and :sob: This is puzzling indeed

**Comment by hmaarrfk** [comment]
it might be related to https://github.com/conda-forge/netcdf-cxx4-feedstock/issues/39

**Comment by akrherz** [comment]
I wish I was more help understanding this failure, ufff.

**Comment by hmaarrfk** [comment]
Some curious posts:
* https://discourse.nixos.org/t/cross-compiling-opencv-from-x86-64-linux-to-aarch64-linux/15446
* https://github.com/opencv/opencv_contrib/issues/2611

**Comment by hmaarrfk** [comment]
Not sure if this is going to create too much noise or not. but you can probably sift through the CMAKE logs to see where HDF is mentionnned.

**Comment by akrherz** [comment]
Thanks, at least we now found the relevant [cmake file](https://github.com/opencv/opencv_contrib/blob/4.x/modules/hdf/CMakeLists.txt), boggling

File: recipe/build.sh [hunk 1]
```diff
@@ -109,6 +109,11 @@ cmake ${CMAKE_ARGS} -LAH -G "Ninja"                                       \
     -DWITH_V4L=$V4L                                                       \
     -DWITH_CUDA=0                                                         \
     -DWITH_CUBLAS=0                                                       \
+    -DHDF5_FOUND=1                                                        \
```
Comment (hmaarrfk) [inline recipe/build.sh:112]: i feel like you want to set a flag like `CMAKE_CROSSCOMPILING` it doesn't seem to appear in the log.

Comment (akrherz) [reply recipe/build.sh:112]: I'm unsure :( please feel free to push something you feel appropriate.

Comment (akrherz) [reply recipe/build.sh]: I finally got podman working locally to test and may have found something that works?  shrug.

**Comment by akrherz** [comment]
Oooo, got green builds on aarch64 and pcc64le, but I suspect os-x will fail.  Have a perhaps final commit coming once I see a failure :)

**Comment by akrherz** [comment]
Sweet, PR is green :)  Look good @hmaarrfk ?

**Comment by hmaarrfk** [comment]
I'm not sure this is actually correct. are the arm ackages actually tested now.

it seems that we are removing some important lines of code that were there for a reason.

i would want somebody with more experience cross compiling for hdf to chime in

**Comment by akrherz** [comment]
That's fair.  My reading of the HDF cmake guard against cross compiling is due to perceived bugs with how cmake attempts to auto-locate HDF.  The comment makes me think that this auto-locate would find system/SDK HDF instead of the build environment HDF.  SInce conda-forge is closely controlling the build environment, I don't think we need to worry about this auto-location/configuration of HDF.  The failed builds before where from HDF not properly compiling, which I think they are now.  Agreed, having somebody more knowledgeable chime in would be great.

**Comment by hmaarrfk** [comment]
can you find in the logs what implementation of HDF is being used? host or build?

**Comment by akrherz** [comment]
Sure @hmaarrfk , [aarch64 python3.7](https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/424364/logs/131) for example,

> 2021-12-07T22:31:12.8167518Z -- HDF5 C compiler wrapper is unable to compile a minimal HDF5 program.
> 2021-12-07T22:31:12.8309639Z -- Found HDF5: $PREFIX/lib/libhdf5.so (found version "1.12.1")

**Comment by hmaarrfk** [comment]
so do we still need anything in hte `host` section?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -58,6 +61,7 @@ requirements:
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - numpy                                  # [build_platform != target_platform]
     - libprotobuf                            # [build_platform != target_platform]
+    - hdf5                                   # [build_platform != target_platform]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:64]: can we try to take this out?

Comment (akrherz) [reply recipe/meta.yaml]: Sure, :eyes: CI

**Comment by akrherz** [comment]
Hehe, I didn't break it!

> Miniforge3-MacOSX-x86_64.sh: line 1: `﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>ServerBusy</Code><Message>Egress is over the account limit.'

**Comment by hmaarrfk** [comment]
I'm not sure it yeilded the results you wish.

From the broken C logs I can see: 
```
C compiler wrapper is unabl
e
 to compile a minimal HDF5 program.
-- Found
 HDF5: $PREFIX/lib/libhdf5.so (
f
ound version "1.12.1") 
```

We can wait to see what happens later. I think the cross compilation issues with HDF should be brought up upstream. The challenge is that sometimes C-make tries to compile (and execute) a working program to check that the compiler is in good shape.

This is somewhat problematic for cross compilation....

**Comment by hmaarrfk** [comment]
Wow it seems to have worked! Nice!

**Comment by hmaarrfk** [comment]
one of the unique aspects of Conda forge is that we "cross compile" but have emulation in Linux that helps utility functions work.

**Comment by akrherz** [comment]
I am not following, is there something more you think I need to look into or we just awaiting somebody else to chime in on this change?

**Comment by hmaarrfk** [comment]
@conda-forge-admin please restart cis

**Comment by hmaarrfk** [comment]
let's let the bot merge. i think this is ok!

**Pull request merged**